### PR TITLE
DB Optimizations

### PR DESCRIPTION
**Pull request by NickM-27**
- Auto vacuum won't defragment the DB but it removes dead pages. For my 1.5 year old DB it reduced the size from 298 MiB to 232MiB
- Optimize queries to remove `!=` which can not use indexes
- Add index for storage calculations
- Increase DB write timeout
- Set synchronous mode to NORMAL
- Increase DB cache

### CODE REVIEW

**Comment by blakeblackshear** [comment]
Let's take a look at how homeassistant does this in core.

**Comment by blakeblackshear** [comment]
Should we add a call to `VACUUM` in this PR or incorporate that later?

**Review by skrashevich — COMMENTED** [review]
`migrator.add_index(Recordings, ('camera', 'segment_size'), unique=False)`

File: migrations/017_update_indexes.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+"""
+import peewee as pw
+
+SQL = pw.SQL
+
+
+def migrate(migrator, database, fake=False, **kwargs):
+    migrator.sql(
+        'CREATE INDEX "recordings_camera_segment_size" ON "recordings" ("camera", "segment_size")'
+    )
```
Comment (skrashevich) [inline migrations/017_update_indexes.py:31]: `migrator.add_index(Recordings, ('camera', 'segment_size'), unique=False)`

File: migrations/017_update_indexes.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def rollback(migrator, database, fake=False, **kwargs):
+    pass
```
Comment (skrashevich) [inline migrations/017_update_indexes.py:35]: `migrator.drop_index(Recordings, ('camera', 'segment_size'))`

File: migrations/017_update_indexes.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    > migrator.rename_table(model, new_table_name)
+    > migrator.add_index(model, *col_names, unique=False)
+    > migrator.drop_index(model, *col_names)
+    > migrator.add_not_null(model, *field_names)
+    > migrator.drop_not_null(model, *field_names)
+    > migrator.add_default(model, field_name, default)
+
+"""
+import peewee as pw
+
```
Comment (skrashevich) [inline migrations/017_update_indexes.py:24]: `from frigate.models import Recordings`

**Comment by NickM-27** [comment]
The changes are causing problems with the tests. I don't think there is an issue doing the indexes manually in SQL especially since this is how all previous indexes have been done as well.

**Comment by blakeblackshear** [comment]
You have to be cautious about using the model objects for tests because the model objects change over time. This can cause situations where historical migrations would be modified.

For example:

1. A field is added to a model in migration 002.
2. That field is later removed from the model in migration 007.

Because you removed the field from the model definition, migration 002 is now broken.

**Comment by blakeblackshear** [comment]
> That's why in the changes I proposed, field names were strings, not objects

It should work then. We ran into some issues when we tried to use the model objects to create new tables in the past.

**Comment by NickM-27** [comment]
The tests were failing on the unique field inside of peewee migrator.py regardless of what I put

**Comment by skrashevich** [comment]
> The tests were failing on the unique field inside of peewee migrator.py regardless of what I put

Can I have more details?

**Comment by NickM-27** [comment]
> > The tests were failing on the unique field inside of peewee migrator.py regardless of what I put
> 
> Can I have more details?

With what was suggested above this error occurs:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/peewee_migrate/router.py", line 194, in run_one
    migrate(migrator, self.database, fake=fake)
  File "<string>", line 31, in migrate
  File "/usr/local/lib/python3.9/dist-packages/peewee_migrate/migrator.py", line 128, in wrapper
    return method(migrator, model, *args, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/peewee_migrate/migrator.py", line 348, in add_index
    field.unique = unique
AttributeError: 'NoneType' object has no attribute 'unique'
```

When I follow the [peewee docs](https://docs.peewee-orm.com/en/latest/peewee/playhouse.html?highlight=migrator#supported-operations), this occurs:

```
[2023-06-08 07:05:37] peewee_migrate                 ERROR   : Migration failed: 017_update_indexes
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/peewee_migrate/router.py", line 194, in run_one
    migrate(migrator, self.database, fake=fake)
  File "<string>", line 31, in migrate
  File "/usr/local/lib/python3.9/dist-packages/peewee_migrate/migrator.py", line 127, in wrapper
    return method(migrator, migrator.orm[model], *args, **kwargs)
KeyError: 'recordings'
```

**Comment by skrashevich** [comment]
> With what was suggested above this error occurs:

Just remove the third option (unique=False) from add_index call

**Review by blakeblackshear — APPROVED** [review]

**Pull request merged**